File: Kite-UI/Kite/model/Message_v2.swift
//
//  Message_v2.swift
//  Kite
//
//  Created by Adam on 7/22/23.
//

import Foundation

struct Message_v2: Codable {
    let id: UUID?
    let msgType: Int
    let from: Int
    let message: String
    let chatId: Int
    var deliveredTime: Date?
    let toUser: Int?
    let viewCount: Int?
    var blob: Data?
    let chatSize: Int?
    
    init(id: UUID, msgType: Int, from: Int, message: String, chatId: Int, deliveredTime: Date?, toUser: Int, viewCount: Int, chatSize: Int) {
        self.id = id
        self.msgType = msgType
        self.from = from
        self.message = message
        self.chatId = chatId
        self.deliveredTime = deliveredTime
        self.toUser = toUser
        self.viewCount = viewCount
        self.blob = nil
        self.chatSize = chatSize
    }
    
    init(id: UUID, msgType: Int, from: Int, message: String, chatId: Int, deliveredTime: Date?, toUser: Int, viewCount: Int, blob: Data, chatSize: Int?) {
        self.id = id
        self.msgType = msgType
        self.from = from
        self.message = message
        self.chatId = chatId
        self.deliveredTime = deliveredTime
        self.toUser = toUser
        self.viewCount = viewCount
        self.blob = blob
        self.chatSize = chatSize
    }
    
    init(id: UUID) {
        self.id = id
        self.msgType = -1
        self.from = -1
        self.message = ""
        self.chatId = -1
        self.deliveredTime = nil
        self.toUser = nil
        self.viewCount = nil
        self.blob = nil
        self.chatSize = 0
    }
}

File: kite-common-pojo/src/main/java/pojo/Message_v2.java
package pojo;

import com.fasterxml.jackson.annotation.JsonFormat;
import org.springframework.data.cassandra.core.mapping.*;

import java.io.Serializable;
import java.nio.ByteBuffer;
import java.sql.Timestamp;
import java.util.Date;
import java.util.UUID;

@Table
public class Message_v2 implements Serializable {

    @PrimaryKey
    @Column
    @CassandraType(type = CassandraType.Name.UUID)
    private UUID id;

    @Column
    private int msgType;

    @Column
    private int from;

    @Column
    private String message;

    @Column
    private int chatId;

    @Column
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX")
    private Timestamp deliveredTime;

    @Column
    private int toUser;

    @Column
    private int viewCount;

    @Column
    private int chatSize;


    public Message_v2(){}

    public Message_v2(UUID id, int from, String message,int msgType, int chatId, Timestamp deliveredTime, int toUser, int viewCount, int chatSize) {
        this.id = id;
        this.from = from;
        this.message = message;
        this.chatId = chatId;
        this.msgType = msgType;
        this.deliveredTime = deliveredTime;
        this.toUser = toUser;
        this.viewCount = viewCount;
        this.chatSize = chatSize;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public int getFrom() {
        return from;
    }

    public void setFrom(int from) {
        this.from = from;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public int getChatId() {
        return chatId;
    }

    public void setChatId(int chatId) {
        this.chatId = chatId;
    }

    public Timestamp getDeliveredTime() {
        return deliveredTime;
    }

    public void setDeliveredTime(Timestamp deliveredTime) {
        this.deliveredTime = deliveredTime;
    }

    public int getToUser() {
        return toUser;
    }

    public void setToUser(int toUser) {
        this.toUser = toUser;
    }

    public int getViewCount() {
        return viewCount;
    }

    public void setViewCount(int viewCount) {
        this.viewCount = viewCount;
    }

    public int getMsgType() {
        return msgType;
    }

    public void setMsgType(int msgType) {
        this.msgType = msgType;
    }

    public int getChatSize() {
        return chatSize;
    }

    public void setChatSize(int chatSize) {
        this.chatSize = chatSize;
    }
}
